src/mscorlib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -1251,7 +1253,15 @@ internal static unsafe int GetIgnoreCaseHash(string source)
                 }
             }
 
-            return Marvin.ComputeHash32(span.AsBytes(), Marvin.DefaultSeed);
+            int hash = Marvin.ComputeHash32(span.AsBytes(), Marvin.DefaultSeed);
+
+            // If input was longer than 250 characters return the borrowed char array.
```

Comment:
Nit: I would just say "Return the borrowed array if necessary" without duplicating the hardcoded constant in the comment.